### PR TITLE
Improve Userstore API getAllUserstore list endpoint with addtional query param called 'attributes'

### PR DESCRIPTION
**Pull request by sarubi**
## Purpose
Resolves https://github.com/wso2/product-is/issues/7811
Depends on https://github.com/wso2/identity-api-server/pull/112

### CODE REVIEW

File: components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.v1/src/gen/java/org/wso2/carbon/identity/api/server/userstore/v1/UserstoresApi.java [hunk 1]
```diff
@@ -130,9 +134,9 @@ public Response getAvailableUserStoreTypes() {
...
     })
-    public Response getSecondaryUserStores(    @Valid@ApiParam(value = "maximum number of records to return")  @QueryParam("limit") Integer limit,     @Valid@ApiParam(value = "number of records to skip for pagination")  @QueryParam("offset") Integer offset,     @Valid@ApiParam(value = "Condition to filter the retrival of records.")  @QueryParam("filter") String filter,     @Valid@ApiParam(value = "Define the order how the retrieved records should be sorted.")  @QueryParam("sort") String sort) {
+    public Response getSecondaryUserStores(    @Valid@ApiParam(value = "maximum number of records to return")  @QueryParam("limit") Integer limit,     @Valid@ApiParam(value = "number of records to skip for pagination")  @QueryParam("offset") Integer offset,     @Valid@ApiParam(value = "Condition to filter the retrival of records.")  @QueryParam("filter") String filter,     @Valid@ApiParam(value = "Define the order how the retrieved records should be sorted.")  @QueryParam("sort") String sort,     @Valid@ApiParam(value = "Define set of user store attributes (as comma separated) to be returned.")  @QueryParam("attributes") String attributes) {
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.v1/src/gen/java/org/wso2/carbon/identity/api/server/userstore/v1/UserstoresApi.java]: let's use requiredAttributes for consistency. We are following that in IDP, APP etc.

Comment (sarubi) [reply components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.v1/src/gen/java/org/wso2/carbon/identity/api/server/userstore/v1/UserstoresApi.java]: addressed

File: components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.v1/src/main/java/org/wso2/carbon/identity/api/server/userstore/v1/core/ServerUserStoreService.java [hunk 2]
```diff
@@ -447,11 +450,37 @@ private UserStoreDTO createUserStoreDTO(UserStoreReq userStoreReq) {
             userStoreList.setSelf(ContextLoader.buildURIForBody(String.format(V1_API_PATH_COMPONENT +
                             UserStoreConstants.USER_STORE_PATH_COMPONENT + "/%s",
                     base64URLEncodeId(jsonObject.getDomainId()))).toString());
+
+            if (StringUtils.isNotBlank(requiredAttributes)) {
+                String[] requiredAttributesArray = requiredAttributes.split(",");
```
Comment (Kanapriya) [inline components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.v1/src/main/java/org/wso2/carbon/identity/api/server/userstore/v1/core/ServerUserStoreService.java]: Shall we move this separator to a constant file?

Comment (sarubi) [reply components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.v1/src/main/java/org/wso2/carbon/identity/api/server/userstore/v1/core/ServerUserStoreService.java]: fixed

**Pull request merged**